### DISCUSSION TITLE
SFDX behind corporate proxy

### DISCUSSION THREAD

**Discussion by AjManoj** [forcedotcom/cli#2375] [Ideas]
Hi 

By default node uses HTTP_PROXY and HTTPS_PROXY environment variables to connect internet behind corporate proxy. But is it possible to directly use proxy server with sfdx?

I am getting below error whenever i have set proxy in windows setting.
`Error: connect ETIMEDOUT 13.112.252.60:443
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1134:16) {
  errno: 'ETIMEDOUT',
  code: 'ETIMEDOUT',
  syscall: 'connect',
  address: '13.112.252.60',
  port: 443
}`

As per  [jsforce corporate proxy bug](https://github.com/jsforce/jsforce/issues/1)

There is [merge request](https://github.com/jsforce/jsforce/pull/581) which let us set corporate proxy and connection is successful. Can we allow setting this paramater as sfdx cli command parameters, which will allow us to use sfdx behind corporate proxy.

[a comment by an automated account is not shown]

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-9607434](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000RJ9aYAG/view)

**Comment by mshanemc** [comment]
Can you help us understand what's not working with the environment variables (ex: HTTP_PROXY) route?

https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_install_cli_firewall.htm

**Comment by AjManoj** [comment]
As we have to switch between mutiple proxies and also sometime switch on switch off proxy, it becomes difficult to actually set global HTTP_PROXY and unset HTTP_PROXY. 

If there is a seperate setting for this available directly from sfdx, then that can be used then it will be easier to configure vscode tasks and as well SFDX vscode plugin can be updated to use vscode settings proxy instead of relying on global HTTP_PROXY

As we can access salesforce both with and wout proxy in our scenerio it impact us

**Reply by ar-srinivas** [reply]
It is good to have an option in every CLI command to pass the proxy settings. Like we pass different parameters in cli command, we can have option to add proxy url.

**Comment by WillieRuemmele** [comment]
https://github.com/forcedotcom/cli/issues/1798
